taskcluster/firefox_tv_taskgraph/job.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import absolute_import, print_function, unicode_literals
+
+from os import path
+
+from six import text_type
+from taskgraph.transforms.job import run_job_using, configure_taskdesc_for_run
+from voluptuous import Required, Schema, Optional
+from pipes import quote as shell_quote
+
+gradlew_schema = Schema({
+    Required("using"): "gradlew",
+    Required("gradlew"): [text_type],
+    Optional("post-gradlew"): [[text_type]],
+    # Base work directory used to set up the task.
+    Required("workdir"): text_type,
+    Optional("use-caches"): bool,
+    Optional("secrets"): [{
+        Required("name"): text_type,
+        Required("path"): text_type,
+        Required("key"): text_type,
+        Optional("json"): bool,
+    }]
+})
+
+@run_job_using("docker-worker", "gradlew", schema=gradlew_schema)
+def configure_gradlew(config, job, taskdesc):
+    run = job["run"]
+    worker = taskdesc["worker"] = job["worker"]
+
+    worker.setdefault("env", {}).update(
+        {"ANDROID_SDK_ROOT": path.join(run["workdir"], "android-sdk-linux")}
+    )
+
+    # defer to the run_task implementation
+    run["command"] = _extract_command(run)
+    secrets = run.pop("secrets", [])
+    scopes = taskdesc.setdefault("scopes", [])
+    scopes.extend(["secrets:get:{}".format(secret(["name"]) for secret in secrets)])
+
+    run["cwd"] = "{checkout}"
+    run["using"] = "run-task"
+    configure_taskdesc_for_run(config, job, taskdesc, job["worker"]["implementation"])
+
+
+def _extract_command(run):
+    pre_gradle_commands = [[
```

Comment:
nit: have you considered the trade-offs of integrating whatever commands you want to run "pre/post-gradle" as a Gradle task? tbh, I'm not sure if it's actually worth it but here's what I came up with:
- Pro: It consolidates our build logic in a single build system/set of scripts
- Pro: It (maybe) makes it easier for folks to run all the build commands
- Pro: if other releng tasks are created in Gradle, you can create task graphs (lol) and dependencies between tasks
- Con: releng might be less familiar with it
- Con: doesn't support other non-Android platforms as well

Reply:
That's a good question! I think that groovy/kotlin code is more maintainable and testable than a shell command. However, this taskgraph-ification is a move to consolidate the automation in our mobile projects, and we used the pre/post-gradle-command mechanism in the Fenix project.
I'm interested in improving this further as we evolve automation here